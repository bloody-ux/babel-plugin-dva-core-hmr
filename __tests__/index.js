/* eslint-disable no-template-curly-in-string */

const pluginTester = require('babel-plugin-tester')
const createBabylonOptions = require('babylon-options')
const plugin = require('../index')

const babelOptions = {
  filename: 'currentFile.js',
  parserOpts: createBabylonOptions({
    stage: 2
  })
}

pluginTester({
  plugin,
  babelOptions,
  snapshot: true,
  tests: {
    'import default app.model': `import HomeModel from '../model/home';
      app.model(HomeModel);
    `,
    'import namepsace app.model': `import { HomeModel } from '../model/home';
    app.model(HomeModel);
    `,
    'require app.model': `const HomeModel = require('../model/home');
    app.model(HomeModel);
    `,
    'require app.model with member': `const HomeModel = require('../model/home').HomeModel;
    app.model(HomeModel);
    `,
    'require app.model inline': `app.model(require('../model/home'));
    `,
    'require app.model inline with member': `app.model(require('../model/home').HomeModel);
    `,
    'app.model with babel options': {
      code: `import HomeModel from '../model/home';
      dva.model(HomeModel)`,
      pluginOptions: { appNames: ['dva'] }
    },
    'should not hit hmr dva.model': `import HomeModel from '../model/home';
    dva.model(HomeModel);
    `,
    'app.model with appImport & appImportName': {
      code: `import React, { Component } from '@alipay/oc/react';
      function doModel({ app }) {
        app.model(require('../model/home').xxxx);
        app.model(require('../model/about'));
      }`,
      pluginOptions: {
        appImport: 'import { getApp } from \'@alipay/oc\';',
        appImportName: 'getApp()'
      }
    }
  }
})

