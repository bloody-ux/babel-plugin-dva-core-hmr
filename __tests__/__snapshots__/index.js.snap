// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app.model with babel options 1`] = `
"
import HomeModel from '../model/home';
      dva.model(HomeModel)

      ↓ ↓ ↓ ↓ ↓ ↓

import HomeModel from '../model/home';
dva.model(HomeModel);

if (module.hot) {
  module.hot.accept(\\"../model/home\\", () => {
    try {
      let newModel = require(\\"../model/home\\");

      if (newModel.default) newModel = newModel.default;
      dva.replaceModel(newModel);
    } catch (e) {
      console.error(e);
    }
  });
}
"
`;

exports[`import default app.model 1`] = `
"
import HomeModel from '../model/home';
      app.model(HomeModel);

      ↓ ↓ ↓ ↓ ↓ ↓

import HomeModel from '../model/home';
app.model(HomeModel);

if (module.hot) {
  module.hot.accept(\\"../model/home\\", () => {
    try {
      let newModel = require(\\"../model/home\\");

      if (newModel.default) newModel = newModel.default;
      app.replaceModel(newModel);
    } catch (e) {
      console.error(e);
    }
  });
}
"
`;

exports[`import namepsace app.model 1`] = `
"
import { HomeModel } from '../model/home';
    app.model(HomeModel);

      ↓ ↓ ↓ ↓ ↓ ↓

import { HomeModel } from '../model/home';
app.model(HomeModel);

if (module.hot) {
  module.hot.accept(\\"../model/home\\", () => {
    try {
      let newModel = require(\\"../model/home\\");

      if (newModel.default) newModel = newModel.default;
      app.replaceModel(newModel);
    } catch (e) {
      console.error(e);
    }
  });
}
"
`;

exports[`require app.model 1`] = `
"
const HomeModel = require('../model/home');
    app.model(HomeModel);

      ↓ ↓ ↓ ↓ ↓ ↓

const HomeModel = require('../model/home');

app.model(HomeModel);

if (module.hot) {
  module.hot.accept(\\"../model/home\\", () => {
    try {
      let newModel = require(\\"../model/home\\");

      if (newModel.default) newModel = newModel.default;
      app.replaceModel(newModel);
    } catch (e) {
      console.error(e);
    }
  });
}
"
`;

exports[`require app.model inline 1`] = `
"
app.model(require('../model/home'));

      ↓ ↓ ↓ ↓ ↓ ↓

app.model(require('../model/home'));

if (module.hot) {
  module.hot.accept(\\"../model/home\\", () => {
    try {
      let newModel = require(\\"../model/home\\");

      if (newModel.default) newModel = newModel.default;
      app.replaceModel(newModel);
    } catch (e) {
      console.error(e);
    }
  });
}
"
`;

exports[`require app.model inline with member 1`] = `
"
app.model(require('../model/home').HomeModel);

      ↓ ↓ ↓ ↓ ↓ ↓

app.model(require('../model/home').HomeModel);

if (module.hot) {
  module.hot.accept(\\"../model/home\\", () => {
    try {
      let newModel = require(\\"../model/home\\");

      if (newModel.default) newModel = newModel.default;
      app.replaceModel(newModel);
    } catch (e) {
      console.error(e);
    }
  });
}
"
`;

exports[`require app.model with member 1`] = `
"
const HomeModel = require('../model/home').HomeModel;
    app.model(HomeModel);

      ↓ ↓ ↓ ↓ ↓ ↓

const HomeModel = require('../model/home').HomeModel;

app.model(HomeModel);

if (module.hot) {
  module.hot.accept(\\"../model/home\\", () => {
    try {
      let newModel = require(\\"../model/home\\");

      if (newModel.default) newModel = newModel.default;
      app.replaceModel(newModel);
    } catch (e) {
      console.error(e);
    }
  });
}
"
`;

exports[`should not hit hmr dva.model 1`] = `
"
import HomeModel from '../model/home';
    dva.model(HomeModel);

      ↓ ↓ ↓ ↓ ↓ ↓

import HomeModel from '../model/home';
dva.model(HomeModel);
"
`;
